### PR TITLE
Upgrade to latest node and more

### PR DESCRIPTION
**Pull request by FinalAngel**
Huge update featuring 
- new webpack version
- new icons system
- new cms related stuff
- ...

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -4,109 +4,148 @@
...
     ],
     sass: [
-        PROJECT_PATH.sass + '/**/*.{scss,sass}'
-    ]
+        PROJECT_PATH.sass + '/**/*.{scss,sass}',
+        '!' + PROJECT_PATH.sass + '/libs/_svgsprite.scss',
+    ],
+    svg: {
+        icons: [PROJECT_PATH.svg + '/icons/**/*.svg'],
+        // other: [PROJECT_PATH.svg + '/other/**/*.svg'],
```
Comment (FinalAngel) [inline gulpfile.js]: I'd remove this

File: gulpfile.js [hunk 2]
```diff
@@ -4,109 +4,148 @@
...
+ * - "gulp sass --debug" (to generate unminified css with sourcemaps)
+ * - "gulp sass:compile" (just generates the base.css out of sass, handy to skip critical css)
+ * - "gulp sass:critical" (splits the base.css with the critical css)
+ * - "gulp sass:rest" (splits the base.css with the remaining "rest" css)
+ * - "gulp sass:inline" (injects the base-critical.css as inline css into the template)
+ */
+// gulp.task('sass', ['sass:critical', 'sass:rest', 'sass:inline']);
+gulp.task('sass', ['sass:compile']);
+gulp.task('sass:compile', task('sass/compile'));
+gulp.task('sass:critical', ['sass:compile'], task('sass/critical'));
```
Comment (FinalAngel) [inline gulpfile.js]: shall we disable this for now then?

File: private/sass/components/_icons.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/* stylelint-disable */
+
+$svg-colors: (
```
Comment (FinalAngel) [inline private/sass/components/_icons.scss:3]: is this still required?

File: private/sass/components/_icons.scss [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    height: 1em;
+
+    svg {
+        display: block;
+        width: 100%;
+        height: 100%;
+    }
+}
+
+@each $key in map-keys($svg-colors) {
```
Comment (FinalAngel) [inline private/sass/components/_icons.scss:31]: thats different now or?

File: private/svg/icons/circle.svg [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (FinalAngel) [inline private/svg/icons/circle.svg:1]: is that used as an example?

File: static/sprites/icons.svg [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="0" height="0" style="position:absolute">
```
Comment (FinalAngel) [inline static/sprites/icons.svg:1]: still required?

Comment (vxsx) [reply static/sprites/icons.svg:1]: i specifically added it

File: private/svg/icons/circle.svg (hunk 5 above)
Comment (vxsx) [reply private/svg/icons/circle.svg:1]: correct

File: private/sass/components/_icons.scss (hunk 3 above)
Comment (vxsx) [reply private/sass/components/_icons.scss:3]: i just added it, so not "still", but it is required

File: private/sass/components/_icons.scss (hunk 4 above)
Comment (vxsx) [reply private/sass/components/_icons.scss:31]: that's the new one

File: gulpfile.js (hunk 2 above)
Comment (vxsx) [reply gulpfile.js]: it is disabled - `gulp sass` would only run compile

File: gulpfile.js (hunk 1 above)
Comment (vxsx) [reply gulpfile.js]: i added it as an example of how to implement additional icon sprites

Comment (FinalAngel) [reply gulpfile.js]: please add a comment then above

File: gulpfile.js (hunk 2 above)
Comment (FinalAngel) [reply gulpfile.js]: please comment this line and add above a comment that this is experimental and should be used with caution

File: private/sass/components/_icons.scss (hunk 3 above)
Comment (FinalAngel) [reply private/sass/components/_icons.scss:3]: (thumbsup)

Comment (FinalAngel) [reply private/sass/components/_icons.scss:3]: 👍

File: private/sass/components/_icons.scss (hunk 4 above)
Comment (FinalAngel) [reply private/sass/components/_icons.scss:31]: 👍

File: private/svg/icons/circle.svg (hunk 5 above)
Comment (FinalAngel) [reply private/svg/icons/circle.svg:1]: :

File: static/sprites/icons.svg (hunk 6 above)
Comment (FinalAngel) [reply static/sprites/icons.svg:1]: 👍

**Pull request merged**